src/jit/importer.cpp

```diff
@@ -4347,11 +4347,14 @@ NamedIntrinsic Compiler::lookupNamedIntrinsic(CORINFO_METHOD_HANDLE method)
 #else // !defined(_TARGET_XARCH_) && !defined(_TARGET_ARM64_)
 #error Unsupported platform
 #endif // !defined(_TARGET_XARCH_) && !defined(_TARGET_ARM64_)
-        else
+        else if (strcmp(methodName, "get_IsSupported") == 0)
         {
-            assert(strcmp(methodName, "get_IsSupported") == 0);
             return NI_IsSupported_False;
         }
+        else
+        {
+            return gtIsRecursiveCall(method) ? NI_Throw_PlatformNotSupportedException : NI_Illegal;
```

Comment:
Another potential benefit of this is that we may not need both the `Isa.cs` and `Isa.PlatformNotSupported.cs` files. We could just have `Isa.cs` and those not supported by the current platform would hit this path (if someone tried to call it without the appropriate check).
* The `Isa.cs` and `Isa.PlatformNotSupported.cs` files are identical, except one is recursive and one explicitly throws.

Reply:
That makes sense, though that would require require all JITs on all targets to handle the intrinsics in this way, right?

Reply:
Yes. It would require all runtimes, once the APIs are supported by that runtime, to consistently handle them.

Reply:
(I think it is probably a post 3.0 cleanup that could happen, after discussing it with Mono/Xamarin and CoreRT/UWP; depending on their support plans for the APIs).